wp-content/themes/workday/inc/widgets/workday-related-posts.php

```diff
@@ -0,0 +1,125 @@
+<?php
+/*
+ * List all of the terms in a custom taxonomy
+ */
+class workday_related_posts_widget extends WP_Widget {
+
+	function __construct() {
+		$widget_ops = array(
+			'classname' 	=> 'workday-related-posts',
+			'description' 	=> __('Lists posts related to the current post', 'largo')
+		);
+		parent::__construct( 'workday-related-posts-widget', __('Workday Related Posts', 'largo'), $widget_ops);
+	}
+
+	function truncate($text, $chars = 120) {
+		if(strlen($text) > $chars) {
+			$text = $text.' ';
+			$text = substr($text, 0, $chars);
+			$text = substr($text, 0, strrpos($text ,' '));
+			$text = $text.'...';
+		}
+		return $text;
+	}
+
+	function widget( $args, $instance ) {
+		global $post;
+		// Preserve global $post
+		$preserve = $post;
+
+		// only useful on post pages
+		if ( !is_single() ) return;
+
+		$title = apply_filters('widget_title', empty( $instance['title'] ) ? __( 'Read Next', 'largo' ) : $instance['title'], $instance, $this->id_base);
+
+		echo $args['before_widget'];
+
+		if ( ! empty( $title ) ) echo $args['before_title'] . $title . $args['after_title'];
+
+		if ( isset( $instance['qty'] ) ) {
+			$related = new Largo_Related( $instance['qty'] );
+		} else {
+			$related = new Largo_Related( 1 );
+		}
+
+		//get the related posts
+		$rel_posts = new WP_Query( array(
+			'post__in' => $related->ids(),
+			'nopaging' => 1,
+			'post__not_in' => array( $post->ID ),
+			'posts_per_page' => ( isset( $instance['qty'] ) ) ? $instance['qty'] : 1,
+			'ignore_sticky_posts' => 1
+		) );
+
+		if ( $rel_posts->have_posts() ) {
+
+			echo '<ul class="related">';
+
+			while ( $rel_posts->have_posts() ) {
+				$rel_posts->the_post();
+				echo '<li>';
+
+				echo '<a class="workday-related-post-thumbnail" href="' . get_permalink() . '">' . get_the_post_thumbnail( get_the_ID(), 'small', array( 'class' => '' ) ) . '</a>';
+
+				largo_maybe_top_term();
+
+				?>
+
+				<h4><a href="<?php the_permalink(); ?>" title="Read: <?php esc_attr( the_title( '','', FALSE ) ); ?>"><?php the_title(); ?></a></h4>
+
+				<?php if ( isset( $instance['show_byline'] ) && $instance['show_byline'] ) { ?>
+					<h5 class="byline">
+						<span class="by-author"><?php largo_byline( true, false, get_the_ID() ); ?></span>
+					</h5>
+				<?php } ?>
+
+				<?php // post excerpt/summary
+				$excerpt = largo_excerpt(get_the_ID(), 2, false, '', true);
+
+				echo '</li>';
```

Comment:
Would it make sense to replace lines 62-77 here with a call to use `partials/widget-content.php`, like in the related posts widget? https://github.com/INN/largo/blob/c41d8519370b48a6e53ce36b3b516d3e54e39d66/inc/widgets/largo-related-posts.php#L48-L66

(see https://github.com/INN/largo/issues/1769)

Reply:
Do you think it'd be better to do it now, or wait until we redo the Largo one and then just use that one once it's updated?

Reply:
If they switched to the largo widget in the future, they'd need to update these styles to refer to that widget. Might be better to let this lie until we update their Largo for that future version of Largo.